src/encoded/static/components/encyclopedia.js

```diff
@@ -1,49 +1,492 @@
+        query.deleteKeyValue('annotation_type');
+        query.deleteKeyValue('annotation_subtype');
+        query.deleteKeyValue('biosample_ontology');
+        // All file options are selected
+        if (newFiles.length === fileOptions.length) {
+            query.addKeyValue('annotation_type', 'representative DNase hypersensitivity sites');
+            query.addKeyValue('annotation_type', 'candidate Cis-Regulatory Elements');
+            query.addKeyValue('biosample_ontology', '*', true);
+        // Only rDHS is selected
+        } else if (newFiles.length === 1 && newFiles[0] === 'rDHS') {
+            query.addKeyValue('annotation_type', 'representative DNase hypersensitivity sites');
+        // rDHS and other annotation subtypes are selected
+        // This is the option that does not currently work *****
+        } else if (newFiles.indexOf('rDHS') > -1) {
+            query.addKeyValue('annotation_type', 'representative DNase hypersensitivity sites');
+            query.addKeyValue('annotation_type', 'candidate Cis-Regulatory Elements');
+            query.addKeyValue('biosample_ontology', '*', true);
+            newFiles.forEach((type) => {
+                if (type !== 'rDHS') {
+                    query.addKeyValue('annotation_subtype', type);
+                }
+            });
+        // Only non-rDHS annotation subtypes are selected
+        } else {
+            newFiles.forEach((type) => {
+                query.addKeyValue('annotation_subtype', type);
+            });
+        }
+        const newHref = query.format();
+        setSelectedFiles(newFiles);
+        setdownloadHref(newHref);
+    };
+
+    // Update assembly, organism, browser files, and download link when user clicks on tab
+    const handleTabClick = (tab) => {
+        setVizFiles([]);
+        setSelectedOrganism(tab);
+        setAssembly(defaultAssemblyByOrganism[tab]);
+
+        // Update download link
+        const query = new QueryString(downloadHref);
+        query.replaceKeyValue('assembly', defaultAssemblyByOrganism[tab]);
+        const newHref = query.format();
+        setdownloadHref(newHref);
+    };
+
+    // Annotation type facet is not a normal facet, at least one annotation must be selected
+    // Facet filters look like radiobuttons
+    const chooseAnnotationType = (type) => {
+        let newAnnotationType = annotationType;
+        if (annotationType.indexOf(type) > -1 && annotationType.length > 1) {
+            newAnnotationType = annotationType.filter((annotation) => annotation !== type);
+        } else if (annotationType.indexOf(type) === -1) {
+            newAnnotationType = [...annotationType, type];
+        }
+        setAnnotationType(newAnnotationType);
+    };
+
+    // Facet filters are redirected to update the urls that populate the browser
+    const handleFacetClick = (e) => {
+        // If clicked filter belongs to the annotation facet, clear filters, or boolean switch, we do not want to prevent default
```

Comment:
Yes, this goes back to the redirecting the links on the default facets. We're using the API to generate the data, but we're not using the links as the url, it's hidden. This all started because we were going to have one link generate one part of the page and another link generate the other part of the page and the API doesn't allow that.